### PR TITLE
CAMEL-11264: Fixed potential NPE in DefaultUndertowHttpBinding

### PR DESCRIPTION
**Pull request by prashant2402**
Please review the fix.

### CODE REVIEW

**Comment by oscerd** [comment]
You van sue ObjectHelper class instead of check != nulla

**Comment by prashant2402** [comment]
@oscerd Committed with ObjectHelper use. however, as per ticket null can be expected in Exception cause so why would we want to throw IllegalArgumentException in case assertion fails?
Please excuse my ignorance.

**Comment by oscerd** [comment]
I mean in the originale if statement you van sue Objecthelper class to check if it is null or not. The original PR was good

**Comment by prashant2402** [comment]
@oscerd I see what you mean, please review my updates, I hope that suffices. Also, added two unit tests to check both cases
Thanks

File: components/camel-undertow/src/main/java/org/apache/camel/component/undertow/DefaultUndertowHttpBinding.java [hunk 1]
```diff
@@ -325,9 +325,7 @@ public Object toHttpResponse(HttpServerExchange httpExchange, Message message) t
                 // we failed due an exception so print it as plain text
                 StringWriter sw = new StringWriter();
                 PrintWriter pw = new PrintWriter(sw);
-                ObjectHelper.notNull(exception.getCause(), "exception cause");
-
-                exception.getCause().printStackTrace(pw);
+                ObjectHelper.applyIfNotEmpty(exception.getCause(), value -> value, () -> exception).printStackTrace(pw);
```
Comment (davsclaus) [inline components/camel-undertow/src/main/java/org/apache/camel/component/undertow/DefaultUndertowHttpBinding.java]: Can we maybe avoid abusing streaming api, and use a != null check. Also I wonder if we should not print the full stack-trace, maybe it should just be `exception.printStackTrace(pw)` instead. I don't recall a reason why we do a getCause. We likely have similar problem in some of the other http components that has this kind of binding code as well.

**Comment by prashant2402** [comment]
Hi @davsclaus, yes, maybe we don't need getCause at all. I've checked other http components and none of them have similar code, they all just print straight from exception. I've committed again undoing last committed tests (not needed if we are just doing exception.printStackTrace(pw))

**Comment by davsclaus** [comment]
@oscerd looks good, you are welcome to merge to branches

**Comment by oscerd** [comment]
Squashed and merged. Thanks @prashant2402 , can you close this? The PR has been merged.

**Comment by oscerd** [comment]
master, 2.19.x and 2.18.x

**Pull request closed without merging**